go-controller/pkg/ovn/egressip.go

```diff
@@ -604,14 +614,30 @@ func (oc *Controller) addEgressNode(egressNode *kapi.Node) error {
 
 func (oc *Controller) deleteEgressNode(egressNode *kapi.Node) error {
 	klog.V(5).Infof("Egress node: %s about to be removed", egressNode.Name)
-	if stdout, stderr, err := util.RunOVNNbctl(
-		"remove",
-		"logical_switch_port",
-		types.EXTSwitchToGWRouterPrefix+types.GWRouterPrefix+egressNode.Name,
-		"options",
-		"nat-addresses=router",
-	); err != nil {
-		klog.Errorf("Unable to remove GARP configuration on external logical switch port for egress node: %s, stdout: %s, stderr: %s, err: %v", egressNode.Name, stdout, stderr, err)
+	lspRes := []nbdb.LogicalSwitchPort{}
+	lsp := nbdb.LogicalSwitchPort{}
+	opModels := []libovsdbops.OperationModel{
+		{
+			ExistingResult: &lspRes,
+			ModelPredicate: func(lsp *nbdb.LogicalSwitchPort) bool {
+				return lsp.Name == types.EXTSwitchToGWRouterPrefix+types.GWRouterPrefix+egressNode.Name
+			},
+
+			DoAfter: func() {
+				lsp = lspRes[0]
+				delete(lsp.Options, "nat-addresses")
+			},
+			ErrNotFound: true,
+		},
+		{
+			Model: &lsp,
+			OnModelUpdates: []interface{}{
+				&lsp.Options,
+			},
+		},
```

Comment:
Why is this operation model needed? You already modify the options field on the logical switch port in the first operation.

Reply:
done